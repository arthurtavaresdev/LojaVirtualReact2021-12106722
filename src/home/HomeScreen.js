import React, {useState, useEffect} from 'react'

import {Link} from 'react-router-dom'

import Header from './Header'
import Footer from './Footer'
import TextoSite from './TextoSite'
import CardCategoria from './CardCategoria'

import './HomeScreen.css'

import * as CategoriaApi from './CategoriaApi'

//import androidImg from './android.png'
//import appleImg from './apple.png'
//import playstationImg from './playstation.png'
//import windowsImg from './windows.png'

export default () => {
  
  const [listaCategorias, setListaCategorias] = useState([])

  useEffect(() => {
    carregarCategorias()
  }, [])

  const carregarCategorias = async () => {
    const categorias = await CategoriaApi.getAll()

    console.log("Executou o useEffect", categorias)
    setListaCategorias(categorias)
  }

  return(
    <div data-testid='home-screen'>
      <Header />
      <section>
        <article>
          <TextoSite />
        </article>
        <aside>
          {
            listaCategorias.map(c =>  
              <Link to={'/categorias/' + c.id}>
                <CardCategoria
                  nome={c.nome}
                  img={c.img128}
                />
              </Link>
            )
          }
        </aside>
      </section>
      <Footer />
    </div>
  )
}