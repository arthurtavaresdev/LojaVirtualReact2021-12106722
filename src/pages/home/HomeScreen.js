import React, {useState, useEffect} from 'react'

import {Link} from 'react-router-dom'

import Header from '../../components/Header'
import Footer from '../../components/Footer'
import TextoSite from '../../components/TextoSIte'
import CardCategoria from '../../components/CardCategoria'

import './HomeScreen.css'

import * as CategoriaApi from '../../services/CategoriaApi'

export default () => {
  
  const [listaCategorias, setListaCategorias] = useState([])

  useEffect(() => {
    carregarCategorias()
  }, [])

  const carregarCategorias = async () => {
    const categorias = await CategoriaApi.getAll()

    console.log("Executou o useEffect", categorias)
    setListaCategorias(categorias)
  }

  return(
    <div data-testid='home-screen'>
      <Header />
      <section>
        <article>
          <TextoSite />
        </article>
        <aside>
          {
            listaCategorias.map(c =>  
              <Link to={'/categorias/' + c.id}>
                <CardCategoria
                  nome={c.nome}
                  img={c.img128}
                />
              </Link>
            )
          }
        </aside>
      </section>
      <Footer />
    </div>
  )
}