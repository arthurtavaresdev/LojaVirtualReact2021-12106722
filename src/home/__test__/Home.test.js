import React from 'react'
import {MemoryRouter} from 'react-router-dom'
import {getByAltText, render, wait} from '@testing-library/react'
import HomeScreen from '../HomeScreen'
import * as CategoriaApi from '../CategoriaApi'

test('Validar fotografia da tela', () => {
    const {getByTestId} = render(<HomeScreen />)
    const div = getByTestId('home-screen')

    expect(div).toMatchSnapshot()
})

test('Quando a API devolver 1 categoria, esta deve ser exibida', async () => {
    CategoriaApi.getAll = jest.fn(() => {
        console.log('Chamou o CategoriaApi.getAll Falso')
        return [{
            id: 1,
            nome: 'Android Teste',
            ativa: true,
            img64: 'http://img64.png',
            img128: 'http://img128.png',
            img256: 'http://img256.png',
        }]
    })

    const {getByText} = render(<HomeScreen />, {wrapper: MemoryRouter})

    await wait(() => getByText('Android Teste'))

    expect(getByText('Android Teste')).toBeInTheDocument()
})