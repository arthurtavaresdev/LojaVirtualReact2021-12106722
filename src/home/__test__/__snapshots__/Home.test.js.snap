// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Validar fotografia da tela 1`] = `
<div
  data-testid="home-screen"
>
  <header
    data-testid="main-header"
  >
    <img
      alt="Logo"
      src="logo.png"
    />
    <h1>
      Loja Virtual
    </h1>
  </header>
  <section>
    <article>
      <p
        class="textoIntrodutorio"
      >
        Texto que introduz o negócio e o serviço para o usuário e permite ao usuário navegar para qualquer uma das quatro categorias de produtos.
      </p>
    </article>
    <aside />
  </section>
  <footer>
    Newton - 2020 - Todos os direitos reservados.
  </footer>
</div>
`;
